lib/resty/kafka/errors.lua

```diff
@@ -1,24 +1,115 @@
 -- Copyright (C) Dejiang Zhu(doujiang24)
 
-
+-- Reference: https://kafka.apache.org/protocol.html#protocol_error_codes
 local _M = {
-    [0]     = 'NoError',
-    [-1]    = 'Unknown',
-    [1]     = 'OffsetOutOfRange',
-    [2]     = 'InvalidMessage',
-    [3]     = 'UnknownTopicOrPartition',
-    [4]     = 'InvalidMessageSize',
-    [5]     = 'LeaderNotAvailable',
-    [6]     = 'NotLeaderForPartition',
-    [7]     = 'RequestTimedOut',
-    [8]     = 'BrokerNotAvailable',
-    [9]     = 'ReplicaNotAvailable',
-    [10]    = 'MessageSizeTooLarge',
-    [11]    = 'StaleControllerEpochCode',
-    [12]    = 'OffsetMetadataTooLargeCode',
-    [14]    = 'OffsetsLoadInProgressCode',
-    [15]    = 'ConsumerCoordinatorNotAvailableCode',
-    [16]    = 'NotCoordinatorForConsumerCode',
+    [0]     = {msg = 'NONE', retriable = false},
+    [-1]    = {msg = 'UNKNOWN_SERVER_ERROR', retriable = false},
+    [1]     = {msg = 'OFFSET_OUT_OF_RANGE', retriable = false},
+    [2]     = {msg = 'CORRUPT_MESSAGE', retriable = true},
+    [3]     = {msg = 'UNKNOWN_TOPIC_OR_PARTITION', retriable = true},
+    [4]     = {msg = 'INVALID_FETCH_SIZE', retriable = false},
+    [5]     = {msg = 'LEADER_NOT_AVAILABLE', retriable = true},
+    [6]     = {msg = 'NOT_LEADER_OR_FOLLOWER', retriable = true},
+    [7]     = {msg = 'REQUEST_TIMED_OUT', retriable = true},
```

Comment:
I'm not should if this is a good idea, but I'm not against it.
Actually, it the kafka server may already receive the message even the client got timeout.
It will change the current resty.kafka library from `at most once` to `at least once`.

If you think the current implementation in this PR could be better, I think we need to document it clearly.

Reply:
Years ago, `-- XX: only 3, 5, 6 can retry`, only these can retry safely, to keep `at most one`.
Again, I'm not against changing it with document it clearly.

Reply:
@doujiang24 I'm not sure about this either, these retryability logs come from kafka's protocol documentation. 🤔

Reply:
okay, let's change it. Align with kafka document.
Please add some doc for message retryable(also refer to kafka doc), otherwise, this PR looks good to me.

Reply:
@bzp2010 Are you still interested in the PR?

Reply:
@doujiang24 Yes, I interested in it still. I will try to adding some documentation.